src/auth/token-verifier.ts

```diff
@@ -69,14 +72,13 @@ export interface FirebaseTokenInfo {
 }
 
 /**
- * Class for verifying general purpose Firebase JWTs. This verifies ID tokens and session cookies.
+ * Class for verifying ID tokens and session cookies.
  */
 export class FirebaseTokenVerifier {
-  private publicKeys: {[key: string]: string};
-  private publicKeysExpireAt: number;
   private readonly shortNameArticle: string;
+  private readonly signatureVerifier: PublicKeySignatureVerifier;
 
-  constructor(private clientCertUrl: string, private algorithm: jwt.Algorithm,
+  constructor(clientCertUrl: string, private algorithm: jwt.Algorithm,
```

Comment:
I did not remove the `algorithm` from here, but we can use the `const ALGORITHM_RS256` directly here instead of passing it around.
We have a unit test: `should not decode a signed token when the algorithm is set to none (emulator)`, which will become obsolete if we use the const.

Reply:
The reason I'm keen on dropping the algorithm parameter from this API is that it's `jsonwebtoken` specific (the type is `jwt.Algorithm`). I think we can just hardcode RS256 into this implementation, and drop the parameter. It's extremely unlikely to change. And of course change/remove any affected tests too.